api/functions/process_file.py

```diff
@@ -0,0 +1,113 @@
+import base64
+import os
+from typing import Any, Dict
+
+from google.cloud import pubsub_v1
+from functions_framework import Context
+from utils.cloud_storage import download_blob, upload_blob, list_blobs_with_prefix
+from utils.firebase import get_firestore_client
+from firebase_admin import firestore
+
+TRANSCRIPTION_EXTENSIONS = ["txt", "eaf"]
+AUDIO_EXTENSIONS = ["wav"]
+
+
+def process_dataset_file(event: pubsub_v1.types.message, context: Context) -> None:
+    """Background Cloud Function which triggers from pubsub dataset-processing
+    topic.
+
+    This processes the incoming file, whose path it can infer from the event. It
+    cleans the data inside if it's a transcription file, and saves it to cloud
+    storage so that it can be used in training with the dataset it came from.
+
+    Parameters:
+         event (dict):  The dictionary with data specific to this type of
+                        event. The `@type` field maps to
+                         `type.googleapis.com/google.pubsub.v1.PubsubMessage`.
+                        The `data` field maps to the PubsubMessage data
+                        in a base64-encoded string. The `attributes` field maps
+                        to the PubsubMessage attributes if any is present.
+         context (google.cloud.functions.Context): Metadata of triggering event.
```

Comment:
I have taken them out in the next revision. I'm interested though to see what you think about this: I still like that when you look at a function with a large number of parameters which include the type beside the parameter in the docstring, that you don't have to hunt for the type at the top of the docstring- what do you think about this?

Reply:
To me it just seems like more work and it introduces the risk of them diverging if people aren't looking out for it when they make changes. But I'm also used to black-enforcement where when params get too long they each get their own line, which makes it easier. But the argument you make is fair enough and this isn't one of those hills I'm willing to die on.

```
def lots_of_params(
  one: str,
  two: int,
  three: bool,
  four: ComplexType
) -> int:
```

Reply:
Actually funny you mention that because the point about them diverging is something I've experienced with the google libraries so that's very interesting. I also don't care either way just wanted to know your opinion :)